x-pack/plugins/security_solution/server/endpoint/services/agent/clients/crowdstrike/crowdstrike_agent_status_client.ts

```diff
@@ -59,6 +67,46 @@ export class CrowdstrikeAgentStatusClient extends AgentStatusClient {
     return keyBy(agentStatusResponse.data?.resources, 'id');
   }
 
+  private async getAgentDetailsFromConnectorAction(
+    agentIds: string[]
+  ): Promise<Record<string, LimitedCrowdstrikeHostDetails>> {
+    try {
+      const getAgentDetails = (await this.connectorActions.execute({
+        params: {
+          subAction: SUB_ACTION.GET_AGENT_DETAILS,
+          subActionParams: {
+            ids: agentIds,
+          },
+        },
+      })) as ActionTypeExecutorResult<CrowdstrikeGetAgentsResponse>;
```

Comment:
`.execute()` is a TS generic -  instead of casting, you can use set the response type on the call to `execute` as: `.execute<CrowdstrikeGetAgentsResponse>(...)`